### PR TITLE
Rianfowler/sc 68110/create hook for preflight endpoints 2

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- refactors preflights check to use hooks 
- updates preflight related components to be functional and typechecked 

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/68110/create-hook-for-preflight-endpoints

#### Special notes for your reviewer:

**Review with whitespace hidden** <-----------------------------

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: web/src/components/apps/AppDetailPage.tsx [hunk 1]
```diff
@@ -514,7 +514,6 @@ function AppDetailPage(props: Props) {
                   render={(renderProps) => (
                     <PreflightResultPage
                       logo={selectedApp.iconUri}
-                      app={selectedApp}
```
Comment (rianfowler) [inline web/src/components/apps/AppDetailPage.tsx:517]: this shows up as an error now that `PreflightResultPage` is typechecked. not sure if it was ever used

File: web/src/components/troubleshoot/PreflightsProgress.tsx [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
+    mysql: "Gathering information about MySQL",
+    postgres: "Gathering information about PostgreSQL",
+    redis: "Gathering information about Redis",
+  };
+  if (name in namesObj) {
+    return namesObj[name];
   }
+
+  return "Gathering details about the cluster";
 }
```
Comment (rianfowler) [inline web/src/components/troubleshoot/PreflightsProgress.tsx:28]: this was in the utilities.js but it's only used here so I moved it out of there to reduce indirection

File: web/src/components/troubleshoot/PreflightsProgress.tsx [hunk 3]
```diff
@@ -63,7 +79,7 @@ export default function PreflightProgress(props) {
           </h1>
           <div className="flex alignItems--center u-marginTop--20">
             <span className="u-fontWeight--bold u-fontSize--normal u-textColor--secondary u-marginRight--10">
-              {percentage + "%"}
+              {percentage > 0 ? `${percentage}%` : "0%"}
```
Comment (rianfowler) [inline web/src/components/troubleshoot/PreflightsProgress.tsx:82]: before `percentage` was a module scoped var that would be updated in a few spots. Got rid of that. 

https://github.com/replicatedhq/kots/blob/74cc0ca690f70946b47584c6813166caac1663ca/web/src/components/troubleshoot/PreflightsProgress.jsx#L21-L43

File: web/src/features/PreflightChecks/api/getPreflightResult.tsx [hunk 4]
```diff
@@ -28,18 +28,69 @@ async function getPreflightResult({
...
 
-  return response;
+  return Boolean(
+    response?.preflightResult?.result?.results?.find(
+      (result) => result?.isFail || result?.isWarn
+    )
+  );
 }
 
 function flattenPreflightResponse({
```
Comment (rianfowler) [inline web/src/features/PreflightChecks/api/getPreflightResult.tsx:96]: will add unit test for this when I fix the TDD stuff in kots.

File: web/src/features/PreflightChecks/api/getPreflightResult.tsx [hunk 5]
```diff
@@ -113,7 +182,7 @@ function useGetPrelightResults({
 
       return getPreflightResult({ slug, sequence });
     },
-    queryKey: ["preflight-results", slug, sequence],
+    queryKey: ["preflight-results", sequence, slug],
```
Comment (rianfowler) [inline web/src/features/PreflightChecks/api/getPreflightResult.tsx:185]: convention i'm using here is [`key`, ...`alphabetizedParameters`]

File: web/src/components/PreflightResultPage.tsx [hunk 6]
```diff
@@ -1,569 +1,284 @@
...
+                type="button"
+                className="btn primary blue"
+                disabled={preflightCheck?.showDeploymentBlocked}
+                onClick={() =>
+                  preflightCheck?.shouldShowConfirmContinueWithFailedPreflights
+                    ? setShowContinueWithFailedPreflightsModal(true)
+                    : deployKotsDownstream({
+                        continueWithFailedPreflights: true,
+                      })
+                }
```
[the remark this answers is not in the document]
Comment (miaawong) [reply web/src/components/PreflightResultPage.tsx:197]: much more readable

**Review by miaawong — APPROVED** [review]
amazing job Rian! We got rid of one repeater! 👏

File: web/src/components/PreflightResultErrors.tsx [hunk 7]
```diff
@@ -1,19 +1,21 @@
...
+import { PreflightResult } from "@src/features/PreflightChecks/types";
 
 type Props = {
+  errors?: string[];
   ignorePermissionErrors: () => void;
   logo: string;
-  preflightResultData?: PreflightResult | null;
-  errors?: PreflightError[];
+  preflightResultData?: PreflightResult[];
+  showRbacError: boolean;
```
[the remark this answers is not in the document]
Comment (miaawong) [reply web/src/components/PreflightResultErrors.tsx:18]: exactly what i was thinking.

**Pull request merged**